### PR TITLE
fix(dispatcher): Move remaining simple cases in apps/ folder to IEven…

### PR DESCRIPTION
**Pull request by nickvergessen**
…tDispatcher

* One step closer for https://github.com/nextcloud/server/pull/38546

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by come-nc — APPROVED** [review]

File: apps/files_sharing/lib/AppInfo/Application.php [hunk 1]
```diff
@@ -148,12 +143,18 @@ public function registerEventsScripts(IEventDispatcher $dispatcher, EventDispatc
 		});
 
 		// notifications api to accept incoming user shares
-		$oldDispatcher->addListener('OCP\Share::postShare', function (OldGenericEvent $event) {
+		$dispatcher->addListener('OCP\Share::postShare', function ($event) {
+			if (!$event instanceof OldGenericEvent) {
+				return;
+			}
```
Comment (marcelklehr) [inline apps/files_sharing/lib/AppInfo/Application.php:149]: Why is this guard needed?

Comment (nickvergessen) [reply apps/files_sharing/lib/AppInfo/Application.php:149]: It helps to trick psalm, so it doesn't yield on `$listener->shareNotification($event)` Later on, which only eats GenericEvent.
But it only is an intermediate step anyway to move forward to 5.4.x of the dispatcher. When we have to update to 6.4.x at the end of the year, GenericEvent will no longer exist and we need to do more work I guess, so I took the soft/low effort approach and added this mini guard.

**Review by marcelklehr — APPROVED** [review]

**Review by susnux — APPROVED** [review]
:broom:

**Comment by nickvergessen** [comment]
Docs in https://github.com/nextcloud/documentation/pull/10858

**Pull request merged**